libbeat/processors/add_docker_metadata/add_docker_metadata.go

```diff
@@ -172,11 +171,15 @@ func (d *addDockerMetadata) Run(event *beat.Event) (*beat.Event, error) {
 		}
 
 		if len(container.Labels) > 0 {
-			labels := common.MapStr{}
+			//dedot first
+			outputLabels := common.MapStr{}
 			for k, v := range container.Labels {
-				safemapstr.Put(labels, k, v)
+				// This is necessary so that ES does not interpret '.' fields as new
+				// nested JSON objects, and also makes this compatible with ES 2.x.
+				label := common.DeDot(k)
```

Comment:
Please, make it optional and disabled by default as in the [`docker` module](https://github.com/elastic/beats/blob/master/metricbeat/module/docker/helper.go#L80).

The plan is to have it consistent along all cases, backport it to 6.x and in a future PR change the defaults to true for 7.0, so it is breaking only then.

Reply:
Thanks! Yes I will make the change in next commit.